File: 백준/Silver/4948. 베르트랑 공준/베르트랑 공준.py
def prime(n):
  if n == 1:
    return False
  for i in range(2, int(n ** 0.5) + 1):
    if n % i == 0:
      return False
  return True

all = list(range(2, 246912))
memo = []

for i in all:
  if prime(i):
    memo.append(i)

while True:
  n = int(input())
  cnt = 0
  if n == 0:
    break
  for i in memo:
    if n < i <= 2 * n:
      cnt += 1
  print(cnt)

File: 백준/Silver/4948. 베르트랑 공준/베르트랑 공준.cc
#include <stdio.h>
#include <iostream>
#include <vector>
#include <cmath>
#include <algorithm>
#include <set>
#include <ostream>
#include <string>
#include <cstring>
#include <sstream>
#include <time.h>
#include <queue>
#include <cstdlib>
#include <stdbool.h>

using namespace std;

#define MAX 123456 * 2 + 1

int n, cnt;
int arr[MAX] = {0, };

void prime(){
    for (int i = 2; i <= MAX; i++){
        arr[i] = i;
    }
    
    for (int i = 2; i <= sqrt(MAX); i++){
        if (arr[i] == 0) continue;
        
        for (int j = i * i; j <= MAX; j += i){
            arr[j] = 0;
        }
    }
    
}

int main(){
    prime();
    
    while(true){
        cnt = 0;
        cin >> n;
        
        if (n == 0) break;
        
        for (int i = n + 1; i <= 2 * n; i++){
            if (arr[i] != 0) cnt++;
        }
        
        cout << cnt << '\n';
    }

    return 0;
}
